src/mesh/RTree.hpp

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <Eigen/Core>
```

Comment:
Not sure but we may not need this since `mesh/Triangle.hpp`

Reply:
I try to state dependencies explicitly instead of relying on transitive includes.
[This is a good read](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf10-avoid-dependencies-on-implicitly-included-names) on why this is a good idea.

Our codebase is a bit of a nightmare in that respect though. Maybe running [includewhatyouuse](https://include-what-you-use.org/ ) on it would be a good first step.

Reply:
Good to know, thank you then 👍